usecases/schema/tenant.go

```diff
@@ -56,18 +64,20 @@ func (m *Manager) AddTenants(ctx context.Context,
 	}
 
 	// create transaction payload
-	partitions, err := m.getPartitions(cls, tenantNames)
+	partitions, err := m.getPartitions(cls, names)
 	if err != nil {
 		return fmt.Errorf("get partitions from class %q: %w", class, err)
 	}
 	request := AddTenantsPayload{
 		Class:   class,
-		Tenants: make([]Tenant, len(partitions)),
+		Tenants: make([]TenantCreate, len(partitions)),
 	}
-	i := 0
-	for name, owners := range partitions {
-		request.Tenants[i] = Tenant{Name: name, Nodes: owners}
-		i++
+	for i, tenant := range tenants {
+		request.Tenants[i] = TenantCreate{
+			Name:   tenant.Name,
+			Nodes:  partitions[tenant.Name],
+			Status: schema.ActivityStatus(tenant.ActivityStatus),
+		}
```

Comment:
current getPartitions implementation returns the same number of partitions as number of tenants provided as input.
wouldn't hurt add len check to be extra safe in case impl changes in the future.